packages/next-app/src/app/stableswapsimulator/(components)/ShareButton.tsx

```diff
@@ -0,0 +1,65 @@
+import { Share1Icon } from "@radix-ui/react-icons";
+import { usePathname } from "next/navigation";
+import { useState } from "react";
+
+import Button from "#/components/Button";
+import { Toast, ToastProvider } from "#/components/Toast";
+import { Tooltip } from "#/components/Tooltip";
+import { useStableSwap } from "#/contexts/StableSwapContext";
+
+export function ShareButton() {
+  const { generateURL } = useStableSwap();
+
+  const onInitialPage = usePathname() === "/stableswapsimulator";
+  const [isNotifierOpen, setIsNotifierOpen] = useState<boolean>(false);
+
+  return (
+    <ToastProvider>
+      <>
+        <Tooltip
+          content={
+            onInitialPage
+              ? "You can only share the analysis page"
+              : "Copy a link with the current simulation parameters"
+          }
+        >
+          <Button shade="dark" disabled={onInitialPage}>
+            <div
+              onClick={() => {
+                if (onInitialPage) return;
+                navigator.clipboard.writeText(generateURL());
+                setIsNotifierOpen(true);
+              }}
+              className="flex flex-row gap-4"
+            >
+              Share simulation
+              <Share1Icon
+                color="white"
+                className="ml-1 font-semibold"
+                height={20}
+                width={20}
+              />
+            </div>
+          </Button>
+        </Tooltip>
+        <Toast
+          content={
+            <div className="flex h-14 flex-row items-center justify-between px-4 py-8">
+              <div className="flex flex-col justify-between space-y-1">
+                <h1 className="text-xl font-medium text-slate12">
+                  Link copied
+                </h1>
+                <h3 className="mb-2 text-sm leading-3 text-slate11">
+                  Use it to access the dashboard state.
+                </h3>
+              </div>
+            </div>
```

Comment:
Had to take a look at this 😅 but the toast was raising an error and the source was not having the content on its component.

This is because on toast we are cloning the props `content` and inserting a new prop called `close`. But your parent tag of content is a `div` and divs [don't accept custom props ](https://reactjs.org/link/attribute-behavior). That's why we have to create a component (local scoped, so inside this page with no export) to wrap up the divs and that component will receive the `close` prop

like this:

```js
      <Toast
        content={<ToastContent />}
        isOpen={isNotifierOpen}
        setIsOpen={setIsNotifierOpen}
        variant={"notification"}
      />
    </>
  );
}

function ToastContent() {
  return (
    <div className="flex h-14 flex-row items-center justify-between px-4 py-8">
      <div className="flex flex-col justify-between space-y-1">
        <h1 className="text-xl font-medium text-slate12">Link copied</h1>
        <h3 className="mb-2 text-sm leading-3 text-slate11">
          Use it to access the dashboard state.
        </h3>
      </div>
    </div>
  );
}
```